sql/moz-fx-data-shared-prod/bigeye_derived/dashboard_service_v1/query.py

```diff
@@ -0,0 +1,169 @@
+from google.cloud import bigquery
+from requests.exceptions import RequestException
+
+# Configuration
+TARGET_PROJECT = os.getenv("TARGET_PROJECT", "moz-fx-data-shared-prod")
+TARGET_TABLE = os.getenv(
+    "TARGET_TABLE", "moz-fx-data-shared-prod.bigeye_derived.dashboard_service"
+)
+BIGEYE_API_KEY = os.environ["BIGEYE_API_KEY"]
+WORKSPACE_IDS = json.loads(os.getenv("WORKSPACE_IDS", "[463, 508, 509]"))
+API_URL = "https://app.bigeye.com/api/v1/dashboard/calculate-data-points-request"
+
+# Data point types to request
+DATA_POINT_TYPES = [
+    "DASHBOARD_DATA_POINT_TYPE_NUMBER_OPEN_ISSUES",
+    "DASHBOARD_DATA_POINT_TYPE_AVERAGE_ISSUE_TIME_TO_CLOSE",
+    "DASHBOARD_DATA_POINT_TYPE_NUMBER_OPEN_ISSUES_BY_COLLECTION",
+    "DASHBOARD_DATA_POINT_TYPE_NUMBER_METRICS_BY_COLLECTION",
+    "DASHBOARD_DATA_POINT_TYPE_NUMBER_ISSUES_CLOSED_BY_USER",
+    "DASHBOARD_DATA_POINT_TYPE_NUMBER_BILLABLE_TABLES",
+    "DASHBOARD_DATA_POINT_TYPE_NUMBER_ISSUES_CLOSED",
+    "DASHBOARD_DATA_POINT_TYPE_NUMBER_METRICS_WITH_ISSUE_BY_CATEGORY",
+    "DASHBOARD_DATA_POINT_TYPE_NUMBER_METRICS_BY_CATEGORY",
+    "DASHBOARD_DATA_POINT_TYPE_PCT_METRICS_WO_ISSUE_BY_CATEGORY",
+    "DASHBOARD_DATA_POINT_TYPE_TOTAL_TABLES",
+    "DASHBOARD_DATA_POINT_TYPE_TOTAL_TABLES_WITH_METRICS",
+    "DASHBOARD_DATA_POINT_TYPE_PCT_TABLES_WITH_METRICS",
+    "DASHBOARD_DATA_POINT_TYPE_TOTAL_TABLES_WITH_METRICS_BY_CATEGORY",
+    "DASHBOARD_DATA_POINT_TYPE_PCT_TABLES_WITH_METRICS_BY_CATEGORY",
+    "DASHBOARD_DATA_POINT_TYPE_TOTAL_TABLES_WITH_ISSUE_BY_CATEGORY",
+    "DASHBOARD_DATA_POINT_TYPE_NUMBER_ISSUES_OPENED",
+    "DASHBOARD_DATA_POINT_TYPE_PCT_ISSUES_INTERACTED",
+    "DASHBOARD_DATA_POINT_TYPE_PCT_TABLES_WITHOUT_ISSUE_BY_CATEGORY",
+    "DASHBOARD_DATA_POINT_TYPE_TOTAL_TABLES_WITHOUT_ISSUE_BY_CATEGORY",
+]
+
+
+def make_api_request(workspace_id: int) -> Dict[str, Any]:
+    """Make API request to Bigeye for a specific workspace."""
+    payload = {
+        "aggregationType": "DASHBOARD_DATE_AGGREGATION_TYPE_DAY",
+        "dataPointTypes": DATA_POINT_TYPES,
+        "workspaceIds": [workspace_id],
+    }
+
+    headers = {
+        "accept": "application/json",
+        "content-type": "application/json",
+        "authorization": BIGEYE_API_KEY,
+    }
+
+    try:
+        response = requests.post(API_URL, json=payload, headers=headers, timeout=30)
+        response.raise_for_status()
+        return response.json()
+    except RequestException as e:
+        raise Exception(f"API request failed for workspace {workspace_id}: {str(e)}")
+
+
+def process_response(response_data: Dict[str, Any]) -> pd.DataFrame:
+    """Process API response into a DataFrame."""
+    try:
+        df = pd.json_normalize(
+            response_data["dataPointSeries"],
+            record_path="dataPoints",
+            meta=["dataPointType"],
+            errors="ignore",
+        )
+        df["workspaceIds"] = df["workspaceIds"].str[0]  # Take first element
+
+        # Drop sourceIds if it exists
```

Comment:
```suggestion
```

I'd avoid explaining what my code does, it'd include a comment if there is some nuance to why we have to do something.